### ISSUE TITLE
Recovered panic in 1.8.1

### ISSUE THREAD

**Issue by alexyao2015** [go-delve/delve#2915]
Please answer the following before submitting your issue:

Note: Please include any substantial examples (debug session output,
stacktraces, etc) as linked gists.
1. What version of Delve are you using (`dlv version`)? 
1.8.1
2. What version of Go are you using? (`go version`)? 
1.17.1
3. What operating system and processor architecture are you using? 
Windows with VScode devcontainer under Alpine on x86
4. What did you do?
Ran the below code with a breakpoint set at the single line in `main()`. This also occurs if the breakpoint is set in code prior to this line of code as well. Simply having `validator.New()` causes the panic as seen in the logs.

I've noticed that this bug does not exist in both master and `1.8.0`. If at all possible, a new release would be great and fixes this issue.

5. What did you expect to see?
No panic.
6. What did you see instead?
Logs below

```
package main

import (
	"github.com/go-playground/validator/v10"
)

func main() {
	validator.New()
}
```

```
2022-02-27T04:59:24Z error layer=dap recovered panic: runtime error: slice bounds out of range [:2] with length 0
goroutine 50 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/go-delve/delve/service/dap.(*Session).recoverPanic(0xc000240000, {0xb0ee60, 0xc000b211d0})
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:559 +0x66
panic({0xa11ce0, 0xc001547c98})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/go-delve/delve/pkg/proc.(*BinaryInfo).openSeparateDebugInfo(0xc000384000, 0xc000a442a0, 0x0, {0xc0001d50d0, 0x1, 0x0})
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/bininfo.go:1213 +0x72f
github.com/go-delve/delve/pkg/proc.loadBinaryInfoElf(0xc000384000, 0xc000a442a0, {0xc001547c50, 0x20}, 0x7f733b4a9000, 0xc000c23ff0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/bininfo.go:1297 +0x20e
github.com/go-delve/delve/pkg/proc.loadBinaryInfo(0xc000384000, 0xc000a565d0, {0xc001547c50, 0x18}, 0xc00152e450)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/bininfo.go:671 +0x16d
github.com/go-delve/delve/pkg/proc.(*BinaryInfo).AddImage(0xc000384000, {0xc001547c50, 0x7f733b53f940}, 0x7f733b4a9000)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/bininfo.go:831 +0x347
github.com/go-delve/delve/pkg/proc/linutil.ElfUpdateSharedObjects({0xb2a5a0, 0xc000248cf0})
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/linutil/dynamic.go:179 +0x1a5
github.com/go-delve/delve/pkg/proc/native.(*nativeProcess).stop(0xc000248cf0, 0xc0001fbf80)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/native/proc_linux.go:589 +0x2c5
github.com/go-delve/delve/pkg/proc/native.(*nativeProcess).ContinueOnce(0xc000248cf0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/native/proc.go:229 +0x13e
github.com/go-delve/delve/pkg/proc.(*Target).Continue(0xc0001ce000)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/pkg/proc/target_exec.go:78 +0x1b1
github.com/go-delve/delve/service/debugger.(*Debugger).Command(0xc000380000, 0xc000165b18, 0xc00023fda0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/debugger/debugger.go:1214 +0xb89
github.com/go-delve/delve/service/dap.(*Session).resumeOnce(0xc000240000, {0xa4c19e, 0x8}, 0xc00023faa0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:3424 +0x225
github.com/go-delve/delve/service/dap.(*Session).resumeOnceAndCheckStop(0xc000240000, {0xa4c19e, 0xc0002400d4}, 0xc000165bd0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:3566 +0x2c
github.com/go-delve/delve/service/dap.glob..func1(0x58, {0xa4c19e, 0x60}, 0x0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:3562 +0x1e
github.com/go-delve/delve/service/dap.(*Session).runUntilStop(0xc000240000, {0xa4c19e, 0x8}, 0xc000165cf8)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:3554 +0x107
github.com/go-delve/delve/service/dap.(*Session).runUntilStopAndNotify(0xc000240000, {0xa4c19e, 0x8}, 0x0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:3435 +0x46
github.com/go-delve/delve/service/dap.(*Session).onConfigurationDoneRequest(0xc000240000, 0xc000b211d0, 0xc00023faa0)
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:1566 +0x305
github.com/go-delve/delve/service/dap.(*Session).handleRequest.func1()
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:686 +0x85
created by github.com/go-delve/delve/service/dap.(*Session).handleRequest
	/go/pkg/mod/github.com/go-delve/delve@v1.8.1/service/dap/server.go:684 +0x10c5
```

**Comment by derekparker** [comment]
Removing the assignee and DAP label after looking a bit more closely at the stack trace.

**Comment by derekparker** [comment]
So you mentioned this does not occur on with the latest code, correct? We will be issuing the next release next Monday, so the fix will go out next week.

**Comment by alexyao2015** [comment]
Yes, that's right. A new release with the latest code would fix this and close this issue.

**Comment by derekparker** [comment]
We've released v1.8.2 which should fix this issue. Please reopen if that's not the case.
